checksum.go

```diff
@@ -93,7 +94,7 @@ func (c *FileChecksum) checksum(source string) error {
 //  <checksum> *file2
 //
 // see parseChecksumLine for more detail on checksum file parsing
-func (c *Client) extractChecksum(u *url.URL) (*FileChecksum, error) {
+func (c *Client) extractChecksum(ctx context.Context, u *url.URL) (*FileChecksum, error) {
```

Comment:
Not in this change, but I'd stay away from the temp directory implementation details in the function comments.

Reply:
Because the implementation could change ? Do you want me to change this ?

Reply:
don't need to change it, more of a general comment -- it's better to document the function use case or specification in a way that users understand it without necessary fully understand the implementation.  Also, when implementation change, comments are very likely to be out of sync.

Reply:
Got it, super true, will try to keep that in mind ! Thanks 🙂